File: 15_observability/src/main/java/com/irayspace/observability/book/BookController.java
package com.irayspace.observability.book;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;


@RestController
@RequestMapping("/book")
public class BookController {

    @Autowired
    private BookService bookService;

    @GetMapping("/{isbn}")
    public Book getByIsbn(@PathVariable String isbn) throws Exception {
        return bookService.getByIsbn(isbn);
    }

}

File: 4_jwt_auth/src/main/kotlin/com/example/demo/book/BookController.kt
package com.example.demo.book

import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.GetMapping

import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.security.core.authority.SimpleGrantedAuthority


@RestController
@RequestMapping("book")
class BookController(val bookService: BookService) {

    @GetMapping
    fun get(): List<String> = bookService.get()
}
